modules/tensor_mechanics/src/materials/LAROMANCEStressUpdateBase.C

```diff
@@ -768,6 +768,31 @@ LAROMANCEStressUpdateBaseTempl<is_ad>::computeResidual(
   for (unsigned int p = 0; p < _num_partitions; p++)
   {
     if (_partition_weights[p])
+    {
+      // compute weight normalizing factor
+      GenericReal<is_ad> weight_normalizer = 0;
+      unsigned int number_of_active_tiles = 0;
+      for (unsigned int t = 0; t < _num_tiles[p]; ++t)
+      {
+        // count number of active tiles
+        number_of_active_tiles += checkInTile(p, t);
+        if (_weights[p][t])
+        {
+          // tile normalization factor (sum of tile weights)
+          weight_normalizer += _weights[p][t];
```

Comment:
At this point, all the tiles with weights should also have a `checkInTile`, right?

Reply:
Not sure if I understand the question. The checkInTile status is not stored anywhere, so a tile does not 'have' or carry a stored checkInTile status. Hence it is checked here whether an input falls inside a tile. The following is important to realize and the reason why the previous tile count was wrong: an input may fall within a tile boundary, but the tile weight may end up to be 0, due to the Sigmoid function. In other words. A tile weight can be zero, while an input still falls within a tile's boundaries.

Reply:
Zero or close to zero? By definition of "weight", if a point is out of the tile, then it should have a weight of zero. If it is in the tile, then it should have a non-zero weight.

Reply:
The Sigmoid function maps the weight to become exactly zero near the tile edges. Try the Sigmoid function : 

1 - 1 / (1 + exp(4*d/(d^2-1))) 

for d = 0.97. It produces exactly 0.0, even though d < 1 (the distance v is a value between -1.0 and 1.0)

Reply:
Sure, zero within machine precision. Makes sense to use `checkInTile` then.